android/shared/src/commonMain/kotlin/com/yeobaek/feature/reader/ReaderViewModel.kt

```
@@ -821,6 +821,28 @@ class ReaderViewModel(
+    fun reportComment(commentId: Long) {
+        if (uiState.reportState is ReportState.Loading) return
+
+        uiState = uiState.copy(reportState = ReportState.Loading)
+
+        viewModelScope.launch {
+            try {
+                commentRepository.reportComment(commentId)
+                uiState = uiState.copy(reportState = ReportState.Success)
+            } catch (e: CancellationException) {
+                throw e
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)